### DISCUSSION TITLE
Way to filter candidates by matching against annotations?

### DISCUSSION THREAD

**Discussion by kings2u** [company-mode/company-mode#1259] [Q&A] [answered]
I want to filter candidates by matching against text in a candidate's `annotation` as well the text of the actual candidate. I realized how convenient it would be, for example, to match against the Yasnippet `name` property of a candidate from the `company-yasnippet` backend (rather than matching only against the Yasnippet `key` property, which is often just an abbreviation that might not be intuitive to remember). Is such a feature possible?

Perhaps this might be implemented for a particular backend by using the `:post-completion` argument in `company-backends`? Maybe, for example, a candidate's annotation could be added to the candidate while matching is taking place and, after a candidate is selected, the annotation part could be deleted by a `:post-completion` function? Or perhaps there's a way to simply ignore the annotation part of the completion candidate when it comes time to insert the completion candidate into the buffer? Any help would be much appreciated!

**Comment by dgutov** [comment, marked as answer]
`post-completion` is triggered at the end, when you have selected the completion to be inserted.

I suppose the easiest way would be to use an alternative matching interface like ivy/helm/counsel, and combine the strings for display (and matching) there. Alternatively, it could be an option in company-mode, with accompanying changes in the matching and rendering functions (we do highlight the part of the completion that search has matched).

But if you wanted to do this at the backend level, this part is trivial:

> Or perhaps there's a way to simply ignore the annotation part of the completion candidate when it comes time to insert the completion candidate into the buffer?

Yes: you delete the last LEN(annotation) characters after insertion, and voila.

Note that any of this is only relevant for non-LSP backends: the protocol makes no provision for separating annotations, so they are always treated as part of the completion (and they're cut away in the end in `post-completion`).

**Reply by kings2u** [reply]
Thank you! I'll cast a vote in favor of an option being added to company-mode, but until then you've given me some useful alternatives :)
